components/core/src/clp/streaming_archive/reader/Archive.cpp

```diff
@@ -38,7 +38,7 @@ void Archive::open(string const& path) {
     archive_format_version_t format_version{};
     try {
         FileReader file_reader{metadata_file_path};
-        ArchiveMetadata const metadata{file_reader};
+        auto const metadata = ArchiveMetadata::create_from_file_reader(file_reader);
```

Comment:
since the FileReader goes out of scope right after the metadata is created, can the method takes metadata_file_path as its parameter and create a FileReader internally?

Reply:
>It will move some of the code (the read and stat) into archive reader, but then this constructor can also be used for sfa.

Is this planned in a future PR? I think it makes sense, but the change you propose might be easier to reason about when we are actually making the changes that include sfa.